### PR TITLE
Implement Web Interface

### LINKED ISSUE #172: Add gradio or streamlit demo for running optimized pipeline

### PR DESCRIPTION
**Pull request by bwook00**
close #172 

- Create web.py
- Added to cli.py to allow web.py to be run from the cli with the `autorag run_web` command
- I made the test code `pytest.mark.skip` because the only way to verify that it works is to go to the web and run it.

[sample yaml file]
- Added a (very simple) best.yaml file. (for testing & I thought it would be nice to have the results of `extract_best_config` in sample_config as well.)
- added config_korean.yaml file for Koreans.

[docs]
- Add tutorial run as Web Interface + Added docs for Web Interface.
- In deploy.py, there was something wrong with sphinx syntax, so I add 1 line.

### CODE REVIEW

**Comment by vkehfdl1** [comment]
> (for testing & I thought it would be nice to have the results of extract_best_config in sample_config as well.) 
This is great idea. I made this to issue #204

File: docs/source/deploy/web.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+You can specify project directory with `--project_dir` option or project_dir parameter.
+```
+```bash
+autorag run_web --yaml_path your/path/to/pipeline.yaml --project_dir your/project/directory
+```
+
+### 2. Use project directory
+```bash
+autorag run_web --yaml_path your/path/to/pipeline.yaml
+```
```
Comment (vkehfdl1) [inline docs/source/deploy/web.md:21]: I think you wanted to mention "Use trial direcetory"
Even the code is totally same with yaml path version. haha:)
문서화 진짜 귀찮구나?

Comment (vkehfdl1) [reply docs/source/deploy/web.md:21]: 제대로안함?

File: requirements.txt [hunk 2]
```diff
@@ -31,3 +31,6 @@ llama-index-embeddings-huggingface
 llama-index-llms-openai
 llama-index-llms-huggingface
 llama-index-llms-openai-like
+
+# WebUI
+streamlit~=1.31.1
```
Comment (vkehfdl1) [inline requirements.txt:36]: delete version limitation if there is no certain version issue.

File: sample_config/config_korean.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
+node_lines:
+- node_line_name: retrieve_node_line
+  nodes:
+  - node_type: retrieval
+    modules:
+    - module_type: vectordb
+      embedding_model: openai
+      top_k: 3
```
Comment (vkehfdl1) [inline sample_config/config_korean.yaml:8]: `top_k` can be node parameter in this config file.

**Review by Eastsidegunn — APPROVED** [review]
I think this is pretty good.
LGTM

File: sample_config/config_korean.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    strategy:
+      metrics:
+      - retrieval_f1
+      - retrieval_recall
+      - retrieval_precision
```
Comment (vkehfdl1) [inline sample_config/config_korean.yaml:13]: How about just use `metrics: [ retrieval_f1, retrieval_recall, retrieval_precision ]`
Is it too long?

**Review by vkehfdl1 — CHANGES_REQUESTED** [review]
+ what is the 'best.yaml' for? 
I don't know what is that configuration file means. 
How about change name to 'extracted_sample.yaml'?? 
And it might be close issue #204 at once. haha:)

**Review by Eastsidegunn — APPROVED** [review]

**Review by vkehfdl1 — APPROVED** [review]
LGTM

**Pull request merged**